packages/field-base/src/input-field-mixin.js

```diff
@@ -0,0 +1,268 @@
+import { ClearButtonMixin } from './clear-button-mixin.js';
+import { DelegateFocusMixin } from './delegate-focus-mixin.js';
+import { FieldAriaMixin } from './field-aria-mixin.js';
+import { InputPropsMixin } from './input-props-mixin.js';
+
+const InputFieldMixinImplementation = (superclass) =>
+  class InputFieldMixinClass extends ClearButtonMixin(FieldAriaMixin(InputPropsMixin(DelegateFocusMixin(superclass)))) {
+    static get properties() {
+      return {
+        /**
+         * Whether the value of the control can be automatically completed by the browser.
+         * List of available options at:
+         * https://developer.mozilla.org/en/docs/Web/HTML/Element/input#attr-autocomplete
+         */
+        autocomplete: {
+          type: String
+        },
+
+        /**
+         * This is a property supported by Safari that is used to control whether
+         * autocorrection should be enabled when the user is entering/editing the text.
+         * Possible values are:
+         * on: Enable autocorrection.
+         * off: Disable autocorrection.
+         */
+        autocorrect: {
+          type: String
+        },
+
+        /**
+         * This is a property supported by Safari and Chrome that is used to control whether
+         * autocapitalization should be enabled when the user is entering/editing the text.
+         * Possible values are:
+         * characters: Characters capitalization.
+         * words: Words capitalization.
+         * sentences: Sentences capitalization.
+         * none: No capitalization.
+         */
+        autocapitalize: {
+          type: String
+        },
+
+        /**
+         * Specify that the value should be automatically selected when the field gains focus.
+         */
+        autoselect: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * The value of the field.
+         */
+        value: {
+          type: String,
+          value: '',
+          observer: '_valueChanged',
+          notify: true
+        }
+      };
+    }
+
+    static get hostProps() {
+      return [...super.hostProps, 'autocapitalize', 'autocomplete', 'autocorrect'];
+    }
+
+    static get observers() {
+      return ['__observeOffsetHeight(errorMessage, invalid, label, helperText)'];
+    }
+
+    /** @protected */
+    get _ariaTarget() {
+      return this._inputNode;
+    }
+
+    /**
+     * @return {HTMLElement | undefined}}
+     * @protected
+     */
+    get inputElement() {
+      return this._inputNode;
+    }
+
+    get focusElement() {
+      return this._inputNode;
+    }
+
+    constructor() {
+      super();
+
+      this._boundOnInput = this._onInput.bind(this);
+      this._boundOnBlur = this._onBlur.bind(this);
+      this._boundOnFocus = this._onFocus.bind(this);
+    }
+
+    /** @protected */
+    connectedCallback() {
+      super.connectedCallback();
+
+      if (this._inputNode) {
```

Comment:
There are some places where we check if `_inputNode` is defined, but there are also other places where we don't. We should commit to one style.

Reply:
Changed to check for `_inputNode`.